### PR TITLE
Various updates to support managed service provider token enhancements

### PR DESCRIPTION
**Pull request by mkeeler**
* The `ACL.Bootstrap` & `ACL.BootstrapTokens` RPC endpoint disabling when msp tokens are configured can now be disabled. Specifically it defaults to allowing those endpoints to remain active and will require tuning the config to enable the "disabling" of those endpoints.
* For agents configured with managed service provider tokens, they will now use the first one of those as the agent token for node registration and coordinate updates.
* Lastly many enterprise only configurations have been moved into the OSS. I got rid of the struct embedding in the top level `agent/config.Config` struct. Instead we now emit warnings for configurations that will not have any effect in Consul outside of the enterprise release. This necessitated the mapstructure update as I found in the lib that was preventing things from working quite right.

### CODE REVIEW

File: agent/config/config.go [hunk 1]
```diff
@@ -179,122 +184,125 @@ type Config struct {
...
+	LogFile                          *string           `json:"log_file,omitempty" hcl:"log_file" mapstructure:"log_file"`
+	LogRotateDuration                *string           `json:"log_rotate_duration,omitempty" hcl:"log_rotate_duration" mapstructure:"log_rotate_duration"`
+	LogRotateBytes                   *int              `json:"log_rotate_bytes,omitempty" hcl:"log_rotate_bytes" mapstructure:"log_rotate_bytes"`
+	LogRotateMaxFiles                *int              `json:"log_rotate_max_files,omitempty" hcl:"log_rotate_max_files" mapstructure:"log_rotate_max_files"`
+	MaxQueryTime                     *string           `json:"max_query_time,omitempty" hcl:"max_query_time" mapstructure:"max_query_time"`
+	NodeID                           *string           `json:"node_id,omitempty" hcl:"node_id" mapstructure:"node_id"`
+	NodeMeta                         map[string]string `json:"node_meta,omitempty" hcl:"node_meta" mapstructure:"node_meta"`
+	NodeName                         *string           `json:"node_name,omitempty" hcl:"node_name" mapstructure:"node_name"`
+	// Enterprise Only
```
Comment (i0rek) [inline agent/config/config.go:253]: nit: putting the comments at the end, wouldn't have changed this whole block.

File: agent/config/config.go [hunk 2]
```diff
@@ -606,21 +617,22 @@ type SOA struct {
...
-	NodeTTL             *string           `json:"node_ttl,omitempty" hcl:"node_ttl" mapstructure:"node_ttl"`
-	OnlyPassing         *bool             `json:"only_passing,omitempty" hcl:"only_passing" mapstructure:"only_passing"`
-	RecursorTimeout     *string           `json:"recursor_timeout,omitempty" hcl:"recursor_timeout" mapstructure:"recursor_timeout"`
-	ServiceTTL          map[string]string `json:"service_ttl,omitempty" hcl:"service_ttl" mapstructure:"service_ttl"`
-	UDPAnswerLimit      *int              `json:"udp_answer_limit,omitempty" hcl:"udp_answer_limit" mapstructure:"udp_answer_limit"`
-	NodeMetaTXT         *bool             `json:"enable_additional_node_meta_txt,omitempty" hcl:"enable_additional_node_meta_txt" mapstructure:"enable_additional_node_meta_txt"`
-	SOA                 *SOA              `json:"soa,omitempty" hcl:"soa" mapstructure:"soa"`
-	UseCache            *bool             `json:"use_cache,omitempty" hcl:"use_cache" mapstructure:"use_cache"`
-	CacheMaxAge         *string           `json:"cache_max_age,omitempty" hcl:"cache_max_age" mapstructure:"cache_max_age"`
-	EnterpriseDNSConfig `hcl:",squash" mapstructure:",squash"`
```
Comment (i0rek) [inline agent/config/config.go:623]: `EnterpriseDNSConfig` is an empty struct which is why it doesn't need to be here. I think we had it here to have fewer conflicts with ent. If we want to remove it here, we could as well get rid of the definition in `agent/config/config_oss.go`

Comment (mkeeler) [reply agent/config/config.go:623]: I got rid of the empty structs.

File: agent/config/config.go (hunk 1 above)
Comment (mkeeler) [reply agent/config/config.go]: I moved all enterprise configurations to the end of the structs and it did reduce the diff quite a bit.

**Pull request merged**